Only in nori-2.6.0_p20211008/: .github
Only in nori-2.6.0: .travis.yml
diff -u -r nori-2.6.0/Gemfile nori-2.6.0_p20211008/Gemfile
--- nori-2.6.0/Gemfile	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/Gemfile	2021-10-08 22:56:58.000000000 +0800
@@ -1,9 +1,6 @@
 source 'https://rubygems.org'
 gemspec
 
-platform :rbx do
-  gem 'json'
-  gem 'racc'
-  gem 'rubysl'
-  gem 'rubinius-coverage'
+if RUBY_VERSION >= "3"
+  gem "rexml", "~> 3.2"
 end
diff -u -r nori-2.6.0/README.md nori-2.6.0_p20211008/README.md
--- nori-2.6.0/README.md	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/README.md	2021-10-08 22:56:58.000000000 +0800
@@ -1,11 +1,9 @@
 Nori
 ====
 
-[![Build Status](https://secure.travis-ci.org/savonrb/nori.png)](http://travis-ci.org/savonrb/nori)
-[![Gem Version](https://badge.fury.io/rb/nori.png)](http://badge.fury.io/rb/nori)
-[![Code Climate](https://codeclimate.com/github/savonrb/nori.png)](https://codeclimate.com/github/savonrb/nori)
-[![Coverage Status](https://coveralls.io/repos/savonrb/nori/badge.png?branch=master)](https://coveralls.io/r/savonrb/nori)
-
+[![CI](https://github.com/savonrb/nori/actions/workflows/test.yml/badge.svg)](https://github.com/savonrb/nori/actions/workflows/test.yml)
+[![Gem Version](https://badge.fury.io/rb/nori.svg)](http://badge.fury.io/rb/nori)
+[![Code Climate](https://codeclimate.com/github/savonrb/nori.svg)](https://codeclimate.com/github/savonrb/nori)
 
 Really simple XML parsing ripped from Crack which ripped it from Merb.  
 Nori was created to bypass the stale development of Crack, improve its XML parser  
diff -u -r nori-2.6.0/lib/nori.rb nori-2.6.0_p20211008/lib/nori.rb
--- nori-2.6.0/lib/nori.rb	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/lib/nori.rb	2021-10-08 22:56:58.000000000 +0800
@@ -22,6 +22,7 @@
       :empty_tag_value               => nil,
       :advanced_typecasting          => true,
       :convert_dashes_to_underscores => true,
+      :scrub_xml                     => true,
       :parser                        => :nokogiri
     }
 
@@ -40,7 +41,7 @@
   end
 
   def parse(xml)
-    cleaned_xml = xml.strip
+    cleaned_xml = scrub_xml(xml).strip
     return {} if cleaned_xml.empty?
 
     parser = load_parser @options[:parser]
@@ -77,4 +78,22 @@
     nil
   end
 
+  def scrub_xml(string)
+    if @options[:scrub_xml]
+      if string.respond_to? :scrub
+        string.scrub
+      else
+        if string.valid_encoding?
+          string
+        else
+          enc = string.encoding
+          mid_enc = (["UTF-8", "UTF-16BE"].map { |e| Encoding.find(e) } - [enc]).first
+          string.encode(mid_enc, undef: :replace, invalid: :replace).encode(enc)
+        end
+      end
+    else
+      string 
+    end
+  end
+
 end
diff -u -r nori-2.6.0/nori.gemspec nori-2.6.0_p20211008/nori.gemspec
--- nori-2.6.0/nori.gemspec	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/nori.gemspec	2021-10-08 22:56:58.000000000 +0800
@@ -12,7 +12,6 @@
   s.description = s.summary
   s.required_ruby_version = '>= 1.9.2'
 
-  s.rubyforge_project = "nori"
   s.license = "MIT"
 
   s.add_development_dependency "rake",     "~> 10.0"
diff -u -r nori-2.6.0/spec/nori/api_spec.rb nori-2.6.0_p20211008/spec/nori/api_spec.rb
--- nori-2.6.0/spec/nori/api_spec.rb	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/spec/nori/api_spec.rb	2021-10-08 22:56:58.000000000 +0800
@@ -108,13 +108,6 @@
       expect(Nori::Parser::Nokogiri).to receive(:parse).and_return({})
       nori.parse("<any>thing</any>")
     end
-
-    it "strips the XML" do
-      xml = double("xml")
-      expect(xml).to receive(:strip).and_return("<any>thing</any>")
-
-      expect(nori.parse(xml)).to eq({ "any" => "thing" })
-    end
   end
 
   context "#parse without :advanced_typecasting" do
@@ -163,8 +156,8 @@
   end
 
   context "#parse with :convert_dashes_to_underscores" do
-    it "can be configured to skip dash to underscope conversion" do
-      xml = '<any-tag>foo bar</any-tag'
+    it "can be configured to skip dash to underscore conversion" do
+      xml = '<any-tag>foo bar</any-tag>'
       hash = nori(:convert_dashes_to_underscores => false).parse(xml)
       expect(hash).to eq({'any-tag' => 'foo bar'})
     end
diff -u -r nori-2.6.0/spec/nori/core_ext/object_spec.rb nori-2.6.0_p20211008/spec/nori/core_ext/object_spec.rb
--- nori-2.6.0/spec/nori/core_ext/object_spec.rb	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/spec/nori/core_ext/object_spec.rb	2021-10-08 22:56:58.000000000 +0800
@@ -5,13 +5,13 @@
   describe "#blank?" do
     [nil, false, [], {}].each do |object|
       it "should return true for: #{object.inspect}" do
-        expect(object.blank?).to be_true
+        expect(object.blank?).to eq true
       end
     end
 
     [true, [nil], 1, "string", { :key => "value" }].each do |object|
       it "should return false for: #{object.inspect}" do
-        expect(object.blank?).to be_false
+        expect(object.blank?).to eq false
       end
     end
   end
diff -u -r nori-2.6.0/spec/nori/nori_spec.rb nori-2.6.0_p20211008/spec/nori/nori_spec.rb
--- nori-2.6.0/spec/nori/nori_spec.rb	2015-05-07 13:36:53.000000000 +0800
+++ nori-2.6.0_p20211008/spec/nori/nori_spec.rb	2021-10-08 22:56:58.000000000 +0800
@@ -28,6 +28,11 @@
         expect(parse(xml)["tag"].strip).to eq("text inside cdata")
       end
 
+      it "should scrub bad characters" do
+        xml = "<tag>a\xfbc</tag>".force_encoding('UTF-8')
+        expect(parse(xml)["tag"]).to eq("a\uFFFDc")
+      end
+
       it "should transform a simple tag with attributes" do
         xml = "<tag attr1='1' attr2='2'></tag>"
         hash = { 'tag' => { '@attr1' => '1', '@attr2' => '2' } }
