diff --git a/emote/__init__.py b/emote/__init__.py
index be84da0..634d6a8 100644
--- a/emote/__init__.py
+++ b/emote/__init__.py
@@ -40,8 +40,6 @@ class EmoteApplication(Gtk.Application):
             self.create_picker_window(True)
             user_data.update_shown_welcome()
 
-        if config.is_flatpak:
-            self.flatpak_autostart()
         self.set_theme()
 
         # Run the main gtk event loop - this prevents the app from quitting
