diff --git a/gdb/remote.c b/gdb/remote.c
index 218bca3..f715dec 100644
--- a/gdb/remote.c
+++ b/gdb/remote.c
@@ -8482,10 +8482,17 @@ remote_target::process_g_packet (struct regcache *regcache)
 
   /* Further sanity checks, with knowledge of the architecture.  */
   if (buf_len > 2 * rsa->sizeof_g_packet)
-    error (_("Remote 'g' packet reply is too long (expected %ld bytes, got %d "
-	     "bytes): %s"),
-	   rsa->sizeof_g_packet, buf_len / 2,
-	   rs->buf.data ());
+    {
+      rsa->sizeof_g_packet = buf_len;
+
+      for (i = 0; i < gdbarch_num_regs (gdbarch); i++)
+	{
+	  struct packet_reg *r = &rsa->regs[i];
+
+	  if (r->pnum != -1)
+	    r->in_g_packet = (r->offset >= rsa->sizeof_g_packet) ? 0 : 1;
+	}
+    }
 
   /* Save the size of the packet sent to us by the target.  It is used
      as a heuristic when determining the max size of packets that the
